knot-thing/src/main/java/com/cesar/knot_sdk/KNoTTypes.kt

```diff
@@ -0,0 +1,129 @@
+    const val KNOT_UNIT_MASS_KG                 =       0x01
+    const val KNOT_UNIT_MASS_G                  =       0x02
+    const val KNOT_UNIT_MASS_LB                 =       0x03
+    const val KNOT_UNIT_MASS_OZ                 =       0x04
+    const val KNOT_UNIT_PRESSURE_PA             =       0x01
+    const val KNOT_UNIT_PRESSURE_PSI            =       0x02
+    const val KNOT_UNIT_PRESSURE_BAR            =       0x03
+    const val KNOT_UNIT_DISTANCE_M              =       0x01
+    const val KNOT_UNIT_DISTANCE_CM             =       0x02
+    const val KNOT_UNIT_DISTANCE_MI             =       0x03
+    const val KNOT_UNIT_DISTANCE_IN             =       0x04
+    const val KNOT_UNIT_ANGLE_RAD               =       0x01
+    const val KNOT_UNIT_ANGLE_DEGREE            =       0x02
+    const val KNOT_UNIT_VOLUME_L                =       0x01
+    const val KNOT_UNIT_VOLUME_ML               =       0x02
+    const val KNOT_UNIT_VOLUME_FLOZ             =       0x03
+    const val KNOT_UNIT_VOLUME_GAL              =       0x04
+    const val KNOT_UNIT_AREA_M2                 =       0x01
+    const val KNOT_UNIT_AREA_HA                 =       0x02
+    const val KNOT_UNIT_AREA_AC                 =       0x03
+    const val KNOT_UNIT_RAIN_MM                 =       0x01
+    const val KNOT_UNIT_DENSITY_KGM3            =       0x01
+    const val KNOT_UNIT_LATITUDE_DEGREE         =       0x01
+    const val KNOT_UNIT_LONGITUDE_DEGREE        =       0x01
+    const val KNOT_UNIT_SPEED_MS                =       0x01
+    const val KNOT_UNIT_SPEED_CMS               =       0x02
+    const val KNOT_UNIT_SPEED_KMH               =       0x03
+    const val KNOT_UNIT_SPEED_MIH               =       0x04
+    const val KNOT_UNIT_VOLUMEFLOW_M3S          =       0x01
+    const val KNOT_UNIT_VOLUMEFLOW_SCMM         =       0x02
+    const val KNOT_UNIT_VOLUMEFLOW_LS           =       0x03
+    const val KNOT_UNIT_VOLUMEFLOW_LM           =       0x04
+    const val KNOT_UNIT_VOLUMEFLOW_FT3S         =       0x05
+    const val KNOT_UNIT_VOLUMEFLOW_GALM         =       0x06
+    const val KNOT_UNIT_ENERGY_J                =       0x01
+    const val KNOT_UNIT_ENERGY_NM               =       0x02
+    const val KNOT_UNIT_ENERGY_WH               =       0x03
+    const val KNOT_UNIT_ENERGY_KWH              =       0x04
+    const val KNOT_UNIT_ENERGY_CAL              =       0x05
+    const val KNOT_UNIT_ENERGY_KCAL             =       0x06
+
+// definition of value type
+    const val KNOT_VALUE_NULL                   =       0x00
+    const val KNOT_VALUE_TYPE_INT               =       0x01
+    const val KNOT_VALUE_TYPE_FLOAT             =       0x02
+    const val KNOT_VALUE_TYPE_BOOL              =       0x03
+    const val KNOT_VALUE_TYPE_RAW               =       0x04
+    const val KNOT_VALUE_TYPE_MIN               =       KNOT_VALUE_TYPE_INT
+    const val KNOT_VALUE_TYPE_MAX               =       KNOT_VALUE_TYPE_RAW + 1
```

Comment:
Why the MAX is not RAW?

Reply:
Since there is a C implementation for this same class, I assume they are checking whether a value is lesser than another, rather than lesser than equal.